tests/php/src/Cli/ReferenceSiteImportCommand.php

```diff
@@ -0,0 +1,390 @@
+	 *                                        arguments.
+	 */
+	private function import_site( SiteDefinition $site_definition, $assoc_args ) {
+		if ( ! Utils\get_flag_value( $assoc_args, 'skip-site-meta', false ) ) {
+			$this->add_result(
+				'Site Meta',
+				( new Import\ImportSiteMeta( $site_definition ) )->process()
+			);
+		}
+
+		foreach ( $site_definition->get_import_steps() as $import_step ) {
+			switch ( $import_step['type'] ) {
+				case 'activate_theme':
+					$this->add_result(
+						'Theme',
+						( new Import\ActivateTheme( $import_step['theme'] ) )->process()
+					);
+					break;
+				case 'activate_plugin':
+					$this->add_result(
+						'Plugin',
+						( new Import\ActivatePlugin( $import_step['plugin'] ) )->process()
+					);
+					break;
+				case 'import_wxr_file':
+					$wxr_path = $import_step['filename'];
+
+					if ( ! path_is_absolute( $wxr_path ) ) {
+						$wxr_path = ReferenceSiteCommandNamespace::REFERENCE_SITES_ROOT . $wxr_path;
+					}
+
+					$this->add_result(
+						'Post',
+						( new Import\ImportWxrFile( $wxr_path ) )->process()
+					);
+					break;
+				case 'import_options':
+					$this->add_result(
+						'Option',
+						( new Import\ImportOptions( $import_step['options'] ) )->process()
+					);
+					break;
+				case 'import_theme_mods':
+					$this->add_result(
+						'Theme Mod',
+						( new Import\ImportThemeMods( $import_step['theme_mods'] ) )->process()
+					);
+					break;
+				case 'import_widgets':
+					$this->add_result(
+						'Widget',
+						( new Import\ImportWidgets( $import_step['widgets'] ) )->process()
+					);
+					break;
+				case 'import_customizer_settings':
+					$this->add_result(
+						'Customizer Setting',
+						( new Import\ImportCustomizerSettings( $import_step['settings'] ) )->process()
+					);
+					break;
+			}
+		}
+
+		$this->print_results( $assoc_args );
+	}
+
+	/**
+	 * Empty the site content.
+	 *
+	 * @param bool $empty_uploads Whether to empty the uploads folder as well.
+	 */
+	private function empty_site( $empty_uploads ) {
+		$command = 'site empty --yes';
+
+		if ( $empty_uploads ) {
+			$command .= ' --uploads';
+		}
+
+		WP_CLI::log(
+			$empty_uploads
+				? 'Emptying the site content and uploads...'
+				: 'Emptying the site content...'
+		);
+
+		WP_CLI::runcommand( $command );
+	}
+
+	/**
+	 * Empty the site's extension folders.
+	 *
+	 * This removes all plugins & themes except for the AMP and the WordPress
+	 * Importer plugins.
+	 */
+	private function empty_extensions() {
+		WP_CLI::log( 'Emptying the site extensions...' );
+
+		$plugins = json_decode(
+			WP_CLI::runcommand(
+				'plugin list --field=name --format=json',
+				[ 'return' => true ]
+			),
+			JSON_OBJECT_AS_ARRAY
+		);
+
+		$plugins = array_filter(
+			$plugins,
+			static function ( $plugin ) {
+				return ! in_array( $plugin, [ 'amp', 'wordpress-importer' ], true );
+			}
+		);
+
+		WP_CLI::runcommand( 'plugin delete ' . implode( ' ', $plugins ) );
```

Comment:
Perhaps safer to deactivate rather than delete?

```suggestion
		WP_CLI::runcommand( 'plugin deactivate ' . implode( ' ', $plugins ) );
```